### PR TITLE
[DONOTMERGE] New packages: nx-libs, perl-Switch, x2goclient & x2goserver

### PR DESCRIPTION
**Pull request by Anachron**
[ci skip]

Everything builds and installs, yet somehow `x2goclient` cannot connect to `x2goserver` and throws the following error:

```
/usr/lib/x2go/x2gocheckport: Zeile 131: 
connect('dbname=/var/lib/x2go/x2go_sessions','',...): 
Syntaxfehler im Ausdruck. (Fehlerverursachendes Zeichen ist \"
('dbname=/var/lib/x2go/x2go_sessions','',...)\"). 
Unable to find free display port or insert new session into database; parameters: port (50), 
hostname (void-lap) and session name (). 
```

If someone wants to chime in and help out I'll be glad.

Edit:
Error comes out from `x2goserver/bin/x2gostartagent` line 270:
```
if [[ "${output}" != 'inserted' ]]; then
  typeset msg="Unable to find free display port or insert new session into datab
  "${X2GO_LIB_PATH}/x2gosyslog" "${0}" 'err' "${msg}"

  # Make x2goclient fail.
  echo "${msg}" >&2
  exit '10'
fi
```

### CODE REVIEW

File: srcpkgs/x2goserver/template [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+maintainer="Anachron <gith@cron.world>"
+license="GPL-3.0"
+homepage="https://wiki.x2go.org/doku.php/wiki:advanced:x2goserver-in-detail"
+distfiles="http://code.x2go.org/releases/source/x2goserver/${pkgname}-${version}.tar.gz"
+checksum="6776aaa354f5a44e349f0b3c176d4988c88a618c2edf46c98a37ae89c069dcd0"
+system_accounts="x2gouser:111 x2goprint:112"
+x2gouser_homedir="/var/lib/x2go"
+x2goprint_homedir="/var/spool/x2go"
+make_dirs="${x2gouser_homedir} 0750 x2gouser
+${x2goprint_homedir} 0750 x2goprint"
```
Comment (maxice8) [inline srcpkgs/x2goserver/template]: ignore the first line and prefix with one whitespace

File: srcpkgs/x2goserver/template [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+build_style=gnu-makefile
+make_build_args="PERL_INSTALLDIRS=vendor"
+makedepends="perl perl-ExtUtils-Helpers"
+depends="perl-DBI perl-DBD-SQLite perl-Config-Simple perl-Capture-Tiny \
+perl-File-BaseDir perl-File-Which perl-Try-Tiny xwininfo setxkbmap dbus \
+lsof psmisc pwgen fuse-sshfs xauth desktop-file-utils shared-mime-info \
+perl-Switch"
+short_desc="OS graphical RD and terminal server based on NX technology"
+maintainer="Anachron <gith@cron.world>"
+license="GPL-3.0"
```
Comment (maxice8) [inline srcpkgs/x2goserver/template]: non-spdx identifier

File: srcpkgs/x2goserver/template [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Template file for 'x2goserver'
+pkgname=x2goserver
+version=4.1.0.3
+revision=1
+#wrksrc="x2goserver-${version}"
```
Comment (maxice8) [inline srcpkgs/x2goserver/template]: remove

File: srcpkgs/x2goserver/template [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+build_style=gnu-makefile
+make_build_args="PERL_INSTALLDIRS=vendor"
+makedepends="perl perl-ExtUtils-Helpers"
+depends="perl-DBI perl-DBD-SQLite perl-Config-Simple perl-Capture-Tiny \
```
Comment (maxice8) [inline srcpkgs/x2goserver/template]: no need for backslash inside a quoted variable

File: srcpkgs/x2goserver/template [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+perl-File-BaseDir perl-File-Which perl-Try-Tiny xwininfo setxkbmap dbus \
+lsof psmisc pwgen fuse-sshfs xauth desktop-file-utils shared-mime-info \
+perl-Switch"
```
Comment (maxice8) [inline srcpkgs/x2goserver/template]: prefix with 1 whitespace

File: srcpkgs/x2goclient/template [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'x2goclient'
+pkgname=x2goclient
+version=4.1.2.1
+revision=1
+build_style=gnu-makefile
+make_build_target="build_client build_man"
+make_install_target="install_client install_man"
+hostmakedepends="qt-devel-tools"
```
Comment (maxice8) [inline srcpkgs/x2goclient/template]: qt4 is EOL

File: srcpkgs/nx-libs/template [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+pkgname=nx-libs
+version=3.5.99.16
+revision=1
+build_style=gnu-makefile
+make_install_args="PREFIX=/usr"
+hostmakedepends="autoconf automake libtool pkg-config"
+makedepends="xorgproto zlib-devel libjpeg-turbo-devel libpng-devel"
+short_desc="NX X11 protocol compression libraries"
+maintainer="Anachron <gith@cron.world>"
+license="GPL-2"
```
Comment (maxice8) [inline srcpkgs/nx-libs/template]: non-spdx identifier

File: srcpkgs/x2goserver/template [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+short_desc="OS graphical RD and terminal server based on NX technology"
+maintainer="Anachron <gith@cron.world>"
+license="GPL-3.0"
+homepage="https://wiki.x2go.org/doku.php/wiki:advanced:x2goserver-in-detail"
+distfiles="http://code.x2go.org/releases/source/x2goserver/${pkgname}-${version}.tar.gz"
+checksum="6776aaa354f5a44e349f0b3c176d4988c88a618c2edf46c98a37ae89c069dcd0"
```
Comment (maxice8) [inline srcpkgs/x2goserver/template]: checksum doesn't need to be quoted

File: srcpkgs/perl-Switch/template [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+wrksrc="${pkgname#*-}-${version}"
+build_style=perl-module
+hostmakedepends="perl"
+depends="perl"
+short_desc="Switch statement for Perl"
+maintainer="Anachron <gith@cron.world>"
+license="GPL-3.0-or-later"
+homepage="https://metacpan.org/pod/Switch"
+distfiles="https://cpan.metacpan.org/authors/id/C/CH/CHORNY/Switch-${version}.tar.gz"
+checksum="31354975140fe6235ac130a109496491ad33dd42f9c62189e23f49f75f936d75"
```
Comment (maxice8) [inline srcpkgs/perl-Switch/template]: doesn't need to be quoted

File: srcpkgs/nx-libs/template [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+pkgname=nx-libs
+version=3.5.99.16
+revision=1
+build_style=gnu-makefile
+make_install_args="PREFIX=/usr"
+hostmakedepends="autoconf automake libtool pkg-config"
+makedepends="xorgproto zlib-devel libjpeg-turbo-devel libpng-devel"
+short_desc="NX X11 protocol compression libraries"
+maintainer="Anachron <gith@cron.world>"
+license="GPL-2.0"
```
Comment (maxice8) [inline srcpkgs/nx-libs/template]: non-spdx identifier

File: srcpkgs/nx-libs/template [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+homepage="http://www.x2go.org"
+distfiles="https://code.x2go.org/releases/source/${pkgname}/${pkgname}-${version}-lite.tar.gz"
+checksum=02c2e382ad27d1b3a0febe31c19f0c2868460f0703b85e8c98ef6203074f7fd4
+
+do_build() {
+  make ${makejobs} PREFIX=/usr \
+  CONFIGURE="./configure ${configure_args} --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --includedir=/usr/include"
```
Comment (maxice8) [inline srcpkgs/nx-libs/template]: please cut with a backslack before it reaches 80 columns

File: srcpkgs/x2goclient/template [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+revision=1
+build_style=gnu-makefile
+make_build_target="build_client build_man"
+make_install_target="install_client install_man"
+hostmakedepends="qt5-tools"
+makedepends="qt5-devel qt5-svg-devel qt5-x11extras-devel libldap-devel libssh-devel libXpm-devel cups-devel"
+depends="nx-libs"
+short_desc="Client for the X2Go remote desktop solution"
+maintainer="Anachron <gith@cron.world>"
+license="GPL-2.0"
```
Comment (maxice8) [inline srcpkgs/x2goclient/template]: non spdx identifier

File: srcpkgs/x2goclient/template [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+homepage="https://wiki.x2go.org/doku.php/doc:usage:x2goclient"
+distfiles="http://code.x2go.org/releases/source/${pkgname}/${pkgname}-${version}.tar.gz"
+checksum="1c18981f0a39929624de9472b0e6c3fa2ac7842837dd306db318af54c5fff2af"
+
+if test -n "${CROSS_BUILD}"; then
```
Comment (maxice8) [inline srcpkgs/x2goclient/template]: use `[ ]` no need for -n

File: srcpkgs/x2goserver/template (hunk 2 above)
Comment (maxice8) [reply srcpkgs/x2goserver/template]: non-spdx identifier

File: srcpkgs/x2goserver/files/x2goserver/run [hunk 14]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+/usr/sbin/x2gocleansessions
+pid=$(cat '/run/x2goserver.pid')
+trap "kill ${pid}" EXIT
+tail --pid="${pid}" -f /dev/null
```
Comment (sirikid) [inline srcpkgs/x2goserver/files/x2goserver/run]: I don’t know how to fix it, but this entire script so against `runit` philosophy.

Comment (Anachron) [reply srcpkgs/x2goserver/files/x2goserver/run]: Yep, it's quite a dirty workaround because x2go automatically forks into the background.

**Pull request closed without merging**